src/adapters/sqlite/makeDispatcher/node/DatabaseBridge.js

```diff
@@ -0,0 +1,374 @@
+
+  // MARK: - Helpers
+
+  toBatchOperations = (operations: any) => {
+    if (typeof operations === 'string') {
+      try {
+        return JSON.parse(operations)
+      } catch (error) {
+        //
+      }
+    }
+    return operations
+  }
+
+  withDriver = (
+    tag: number,
+    resolve: any => void,
+    reject: any => void,
+    functionName: string,
+    action: (driver: DatabaseDriver) => any,
+  ) => {
+    try {
+      const connection = this.connections[tag]
+      if (!connection) {
+        throw new Error(`No driver for with tag ${tag} available`)
+      }
+      if (connection.status === 'connected') {
+        if (connection.synchronous) {
+          throw new Error(`Can't perform async action on synchronous connection ${tag}`)
+        }
+        const result = action(connection.driver)
+        resolve(result)
+      } else if (connection.status === 'waiting') {
+        // consoleLog('Operation for driver (tagID) enqueued')
+        // try again when driver is ready
+        connection.queue.push(() => {
+          this.withDriver(tag, resolve, reject, functionName, action)
+        })
+      }
+    } catch (error) {
+      this.sendReject(reject, error, functionName)
+    }
+  }
+
+  synchronously = (functionName: string, action: () => any) => {
+    try {
+      const result = action()
+      return { status: 'success', result }
+    } catch (error) {
+      return { status: 'error', code: `db.${functionName}.error`, message: error.message }
+    }
+  }
+
+  withDriverSynchronous = (
+    tag: number,
+    functionName: string,
+    action: (driver: DatabaseDriver) => any,
+  ) => {
+    return this.synchronously(functionName, () => {
+      const connection = this.connections[tag]
+      if (!connection) {
+        throw new Error(`No or invalid connection for tag ${tag}`)
+      }
+      const actionResult = action(connection.driver)
+      return actionResult
+    })
+  }
+
+  connectDriverAsync = (tag: number, driver: DatabaseDriver) => {
+    const { queue = [] } = this.connections[tag]
+    this.connections[tag] = { driver, synchronous: false, queue: [], status: 'connected' }
+
+    queue.forEach(operation => operation())
+  }
+
+  disconnectDriver = (tag: number) => {
+    const { queue = [] } = this.connections[tag]
+    delete this.connections[tag]
+
+    queue.forEach(operation => operation())
+  }
+
+  assertNoConnection = (tag: number) => {
+    if (this.connections[tag]) {
+      throw new Error(`A driver with tag ${tag} already set up`)
+    }
+  }
+
+  sendReject = (reject: (string, string, Error) => void, error: Error, functionName: string) => {
+    if (reject) {
+      reject(`db.${functionName}.error`, error.message, error)
+    } else {
+      throw new Error(`db.${functionName} missing reject (${error.message})`)
+    }
+  }
+}
+
+const databaseBridge = new DatabaseBridge()
+
+export default databaseBridge
```

Comment:
TBH I don't understand why you went through all the pain and effort to translate Swift `async` and `synchronous` methods given that there is no distinction in node and all methods are synchronous anyway. But OK, I'll accept it.

Reply:
Easy: I had no idea on HOW it gonna behave, I needed it working fast. So I did it "duplicated" and later I would improve it as needed. It was much more `translation` than `coding something new`

Reply:
fair enough